libhdf5/hdf5type.c

```diff
@@ -52,20 +52,17 @@ NC4_inq_type_equal(int ncid1, nc_type typeid1, int ncid2,
    if ((typeid1 <= NC_STRING && typeid2 > NC_STRING) ||
        (typeid2 <= NC_STRING && typeid1 > NC_STRING))
    {
-      if (equalp) *equalp = 0;
```

Comment:
This is a good catch, thanks!

I would actually rewrite this so that a NULL equalp does not return NC_NOERR immediately. That's not how our other inq functions work. Normally, even if the user provides a NULL where we would put the answer, we still go through all the work before not doing anything. ;-)

That is, for this function, I would expect that even if I provided NULL, I would get NC_EBADTYPE if one of the types is not found.

The way to do this is to have a local variable "equal" and then, at the end of the function, set equalp if it was provided.